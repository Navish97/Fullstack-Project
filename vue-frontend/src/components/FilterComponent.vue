<template>
    <div class = "filter-wrapper">
        <div class = "price">
            <label>Price</label>
            <div class = "price-input">
                <div>
                    <input type="number" id="min-price" name ="min-price">
                    <label for="min-price">From</label>
                </div>
                <div>
                    <input type="number" id="max-price" name ="max-price">
                    <label for="max-price">To</label>
                </div>
            </div>
        </div>
    </div>
</template>

<script setup lang="ts">
</script>


<style scoped>
    .filter-wrapper{
        width: 200px;
        background-color: rgba(30, 29, 29, 0.99);
        color: #ffffff;
        padding: 12px 16px;
        border-radius: 4px;
        border: 1px solid #646464;
        font-size: 14px;
        cursor: pointer;
        text-align: left;
        align-content: center;
    }
    .price-input{
        display: inline-flex;
        margin: 3%;
    }
    .price-input input{
        width: 97%;
    }

</style>