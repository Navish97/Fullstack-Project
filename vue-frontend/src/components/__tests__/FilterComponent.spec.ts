import {beforeEach, expect, it} from "vitest";
import {createPinia} from "pinia";
import {mount} from "@vue/test-utils";
import FilterComponent from "@/components/Filter/FilterComponent.vue";


let wrapper: any;

beforeEach(() => {

    const pinia = createPinia();

    wrapper = mount(FilterComponent, {
        global: {
            plugins: [pinia],
        },
    });
});

it("it renders the button properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
});


