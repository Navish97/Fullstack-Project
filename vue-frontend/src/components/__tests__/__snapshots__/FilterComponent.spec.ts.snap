// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it renders the button properly 1`] = `
"<div data-v-2abb9aef=\\"\\" class=\\"filter-wrapper\\">
  <div data-v-2abb9aef=\\"\\" class=\\"header\\">Filter</div>
  <div data-v-2abb9aef=\\"\\" class=\\"price\\"><label data-v-2abb9aef=\\"\\">Price</label>
    <div data-v-2abb9aef=\\"\\" class=\\"price-input\\">
      <div data-v-2abb9aef=\\"\\"><label data-v-2abb9aef=\\"\\" for=\\"min-price\\">From</label><input data-v-2abb9aef=\\"\\" type=\\"number\\" id=\\"min-price\\" name=\\"min-price\\"></div>
      <div data-v-2abb9aef=\\"\\"><label data-v-2abb9aef=\\"\\" for=\\"max-price\\">To</label><input data-v-2abb9aef=\\"\\" type=\\"number\\" id=\\"max-price\\" name=\\"max-price\\"></div>
    </div>
  </div>
  <div data-v-2abb9aef=\\"\\" class=\\"category\\"><label data-v-2abb9aef=\\"\\">Category</label><select data-v-2abb9aef=\\"\\">
      <option data-v-2abb9aef=\\"\\" value=\\"\\">All</option>
    </select></div><button data-v-2abb9aef=\\"\\" class=\\"apply\\">Apply</button><button data-v-2abb9aef=\\"\\" class=\\"reset\\">Reset</button>
</div>"
`;
